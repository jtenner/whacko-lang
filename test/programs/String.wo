export fn _start(): void {
  let a = "Hello world";
  console.log(a);
}