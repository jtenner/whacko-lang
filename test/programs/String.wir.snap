program:
  callable:[std/builtins.wo]gc.initialize;functionType:()=>voidType
(extern function)
  callable:[std/builtins.wo]gc.alloc;functionType:(u32Type,usizeType)=>ptrType
(extern function)
  callable:[std/builtins.wo]gc.store_field_ptr;functionType:(i32Type,ptrType,ptrType,ptrType)=>voidType
(extern function)
  callable:[test/programs/String.wo]_start;functionType:()=>voidType
    block: entry~567
      568~inst = (str "Hello world")
      569~inst = store: (variable a) = (runtime classType:[std/String.wo]String<> 568~inst)
      570~inst = load: (variable a)
      571~inst = store: (variable compiler-generated) = (runtime classType:[std/String.wo]String<> 570~inst)
      574~inst = call: callable:[std/console.wo]console.log;functionType:(classType:[std/String.wo]String<>)=>voidType with (runtime classType:[std/String.wo]String<> 570~inst)
      return (void)

  callable:[std/console.wo]console.log;functionType:(classType:[std/String.wo]String<>)=>voidType
(extern function)
